### PR TITLE
Add 2d view compatible VU

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Adds `VUID-VkImageCreateInfo-flags-07755`

### CODE REVIEW

**Review by arno-lunarg — APPROVED** [review]
LGTM!

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM / +2

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -5556,9 +5556,8 @@ bool CoreChecks::ValidateImageSubresourceRange(const uint32_t image_mip_count, c
...
                                                          const VkImageSubresourceRange &subresourceRange) const {
     bool is_khr_maintenance1 = IsExtEnabled(device_extensions.vk_khr_maintenance1);
-    bool is_2d_compatible = (image_state->createInfo.flags & VK_IMAGE_CREATE_2D_ARRAY_COMPATIBLE_BIT) ? true : false;
-    if (device_extensions.vk_ext_image_2d_view_of_3d)
-        is_2d_compatible |= (image_state->createInfo.flags & VK_IMAGE_CREATE_2D_VIEW_COMPATIBLE_BIT_EXT) ? true : false;
+    bool is_2d_compatible =
+        image_state->createInfo.flags & (VK_IMAGE_CREATE_2D_ARRAY_COMPATIBLE_BIT | VK_IMAGE_CREATE_2D_VIEW_COMPATIBLE_BIT_EXT);
     bool is_image_slicable =
             (image_state->createInfo.imageType == VK_IMAGE_TYPE_3D) && is_2d_compatible;
     bool is_3_d_to_2_d_map = is_khr_maintenance1 && is_image_slicable && is_imageview_2d_type;
```
Comment (juan-lunarg) [inline layers/buffer_validation.cpp:5563]: const where applicable

**Review by juan-lunarg — COMMENTED** [review]
minor const correctness nitpick

File: layers/buffer_validation.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply layers/buffer_validation.cpp:5563]: agree :100: , but going to make separate PR now that does this everywhere!

**Pull request merged**